### PR TITLE
Using clang

### PR DESCRIPTION
**Pull request by roigcarlo**
Do not merge.

In my quest to reduce build times I shall try the new clang 5.0 that came with the newest trusty image.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,17 +15,15 @@ env:
...
   apt:
     sources:
       - sourceline: 'ppa:ubuntu-toolchain-r/test'
       - sourceline: 'ppa:boost-latest/ppa'
     packages:
-      - gcc-6
-      - g++-6
-      - gfortran-6
+      - gfortran
       - libboost1.55-dev
```
Comment (RiccardoRossi) [inline .travis.yml:27]: why so old? i think on the new distro they should have a newer boost by default

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 sudo: false
 dist: trusty
 language: cpp
-compiler: g++
+compiler: clang
```
Comment (RiccardoRossi) [inline .travis.yml:4]: just out of curiosity: the new image doesn't have gcc7 by default? telling as i think gcc is currently faster than clang

File: scripts/build/travis/configure_travis_trusty.sh [hunk 3]
```diff
@@ -82,8 +82,8 @@ CXX_WARN_FLAGS="-Wall"
 #        C_CUSTOM_FLAGS="-fmessage-length=20"
 #        CXX_CUSTOM_FLAGS="-fmessage-length=20"
 # --------------------------------------------------------------------------------------------------------------
-C_CUSTOM_FLAGS=""
-CXX_CUSTOM_FLAGS=""
+C_CUSTOM_FLAGS="-flto"
```
Comment (RiccardoRossi) [inline scripts/build/travis/configure_travis_trusty.sh:85]: don't you want to use "thin-lto" instead? this takes an enormous amount of time and of memory

**Comment by roigcarlo** [comment]
They are using 14.04. According to them, defaults are 4.8 ( not 4.8.5). Clang for some reason is not 3.X. Boost is not event installed so I have to chose one.
I will drop this, was only to check if we could compile using system's but since not even clang works without gold, no reason for this to be

**Comment by RiccardoRossi** [comment]
they just updated the version! see the link i passed you yesterday

**Comment by roigcarlo** [comment]
The one you sent me is from amazon linux, they use another image...

**Pull request closed without merging**